iroh/src/api.rs

```diff
@@ -0,0 +1,93 @@
+use std::collections::{HashMap, HashSet};
+use std::path::Path;
+
+use anyhow::Result;
+use async_trait::async_trait;
+use bytes::Bytes;
+use cid::Cid;
+use iroh_resolver::resolver::Path as IpfsPath;
+use libp2p::gossipsub::MessageId;
+use libp2p::{Multiaddr, PeerId};
+
+pub struct Id {
+    pub peer_id: PeerId,
+    pub listen_addrs: Vec<Multiaddr>,
+    pub local_addrs: Vec<Multiaddr>,
+}
+
+#[derive(Debug)]
+pub enum Ping {
+    PeerId(PeerId),
+    Multiaddr(Multiaddr),
+}
+
+pub trait Api<P: P2p, S: Store>: Accessors<P, S> + GetAdd {}
+
+pub trait Accessors<P: P2p, S: Store> {
+    fn p2p(&self) -> Result<P>;
+    fn store(&self) -> Result<S>;
+}
+
+#[async_trait(?Send)]
+pub trait GetAdd: std::marker::Sync + std::marker::Send {
+    async fn get(&self, ipfs_path: &IpfsPath, output_path: Option<&Path>) -> Result<()>;
+    async fn add(&self, path: &Path, recursive: bool, no_wrap: bool) -> Result<Cid>;
+}
+
+impl<T: Accessors<P, S> + GetAdd, P: P2p, S: Store> Api<P, S> for T {}
+
+#[async_trait]
+pub trait P2pConnectDisconnect {
+    async fn connect(&self, peer_id: &PeerId, addrs: &[Multiaddr]) -> Result<()>;
+    async fn disconnect(&self, peer_id: &PeerId) -> Result<()>;
+}
+
+#[async_trait]
+pub trait P2pId: std::marker::Sync {
```

Comment:
`Send` and `Sync` are auto imported, no need to add `std::marker::` infront of them